rbac/management/utils.py

```diff
@@ -202,8 +202,10 @@ def roles_for_cross_account_principal(principal):
     """Return roles for cross account principals."""
     target_account, user_id = principal.username.split("-")
     with tenant_context(Tenant.objects.get(schema_name="public")):
-        cross_account_request = CrossAccountRequest.objects.filter(
+        cross_account_requests = CrossAccountRequest.objects.filter(
             target_account=target_account, user_id=user_id, status="approved"
-        ).first()
-        role_names = list(cross_account_request.roles.all().values_list("name", flat=True))
+        )
+        role_names = []
+        for car in cross_account_requests:
+            role_names.extend(list(car.roles.all().values_list("name", flat=True)))
```

Comment:
This will return duplicate names if the same role exists in multiple CARs, currently. Probably not a big deal since the filter below will be unique, but wanted to mention it.

I believe we could actually do something like this, and replace:
```
        cross_account_requests = CrossAccountRequest.objects.filter(
            target_account=target_account, user_id=user_id, status="approved"
        )
        role_names = []
        for car in cross_account_requests:
            role_names.extend(list(car.roles.all().values_list("name", flat=True)))
    return Role.objects.filter(name__in=role_names)
```
with this:
```
        role_names = CrossAccountRequest.objects.filter(
            target_account=target_account,
            user_id=user_id,
            status="approved"
        ).values_list("roles__name", flat=True).distinct()
    return Role.objects.filter(name__in=list(role_names))
```
Would need to be tested, though. Just an idea. :)

Reply:
Nice! Updated